joomla/joomla-cms
# [5.4] Add attribute to fix B/C issue in list form field

Pull Request for Issue #44484  .

### Summary of Changes
with the merging of https://github.com/joomla/joomla-cms/pull/43842 the functionality that was present in Joomla for years has changed for extensions using that functionality.
because there is no way to restore the used and required functionality this can be considered a B/C.
Furthermore the PR introduced ambiguity as it changed the logic / functionality only for one field.

See discussion here: #44484 

This PR adds an attribute with which a developer can 'restore' the showon functionality as it was, at least giving them the option to keep using the core ListField instead of developing a workaround.

this PR adds an form attribute: showonlocal="true/false" (default = false)

when setting showonlocal, the showon will acto on the local set value (like it has been for years in pré 5.2), when not set or when set to false, showon will act on the global value (like it is currently in 5.2)

pinging @LadySolveig 

### Testing Instructions

Set Global Configuration -> Contacts: Options -> Tab: Contact Information --> Show

Open a contact in edit mode or add a new contact.

Set Contacts: Edit -> Tab: Display -> Contact Information --> Use Global (Show)

<Test Results 1>

open file ./administrator/component/com_contact/forms/contact.xml
add attribute showonLocal="false" to field "show_info"
```
				<field
					name="show_info"
					type="list"
					label="COM_CONTACT_FIELD_SHOW_INFO_LABEL"
					class="form-select-color"
					useglobal="true"
					validate="options"
					showonLocal="false"
					>
					<option value="0">JHIDE</option>
					<option value="1">JSHOW</option>
				</field>
```
<test results should be the same as Test 1>

change showonLocal to "true"

<Test Results 2>

in Contact Edit, Display tab, toggle Contact Information from "Use Global" to "Show"

<Test Results 3>

### Actual result BEFORE applying this Pull Request
Contact Edit , Display Tab Fields:
- Name
- Contact's Position

are always displaying

### Expected result AFTER applying this Pull Request

## Test Results 1
Contact Edit , Display Tab Fields:
- Name
- Contact's Position

are displaying

## Test Results 2
Contact Edit , Display Tab Fields:
- Name
- Contact's Position

are hidden

## Test Results 3
Contact Edit , Display Tab Fields:
- Name
- Contact's Position

are displaying


### Link to documentations
Please select:
